### PR TITLE
Add dynamic template support to Twilio Sendgrid binding

### PR DESCRIPTION
**Pull request by yannart**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [x] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [x] Commands include options for Linux, MacOS, and Windows within codetabs
- [x] New file and folder names are globally unique
- [x] Page references use shortcodes instead of markdown or URL links
- [x] Images use HTML style and have alternative text
- [x] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description
Documentation for the PR: https://github.com/dapr/components-contrib/pull/2844/

Adding support for dynamic templates. This includes allowing to provide a template id and template JSON data.

## Issue reference
https://github.com/dapr/docs/issues/3421
https://github.com/dapr/components-contrib/issues/2412

### CODE REVIEW

**Comment by msfussell** [comment]
@yannart - Is this targeting the v1.12 release ( in the code). If so then can you point this to the v1.12 docs branch?

File: daprdocs/content/en/reference/components-reference/supported-bindings/sendgrid.md [hunk 1]
```diff
@@ -78,6 +83,19 @@ You can specify any of the optional metadata properties on the output binding re
 }
 ```
 
+If a dynamic template is used, the template id needs to be used and if provided, the template data is used:
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/sendgrid.md]:
```suggestion
## Dynamic templates
If a dynamic template is used, a `dynamicTemplateId` needs to be provided and then the`dynamicTemplateData` is used:
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/sendgrid.md [hunk 2]
```diff
@@ -55,7 +59,8 @@ The above example uses secrets as plain strings. It is recommended to use a secr
 | emailCc | N | Output | If set this specifies the 'cc' email address of the email message. Only a single email address is allowed. Optional field, see [below](#example-request-payload) | `"me@example.com"` |
 | emailBcc | N | Output | If set this specifies the 'bcc' email address of the email message. Only a single email address is allowed. Optional field, see [below](#example-request-payload) | `"me@example.com"` |
 | subject | N | Output | If set this specifies the subject of the email message. Optional field, see [below](#example-request-payload) | `"subject of the email"` |
-
+| dynamicTemplateId | N | Output | If set this specifies the dynamic template id. Optional field, see [below](#example-request-payload) | `"d-123456789"` |
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/sendgrid.md:62]: Can dynamicTemplateId be any user defined value? Where does this Id come from, is this created by the user and can have any string format? Or is there a defined, format for the Id?

Comment (yannart) [reply daprdocs/content/en/reference/components-reference/supported-bindings/sendgrid.md:62]: It is the Sendgrid template id which is a string. The format is not well documented (see example of API here: https://docs.sendgrid.com/api-reference/transactional-templates-versions/retrieve-a-specific-transactional-template-version)

File: daprdocs/content/en/reference/components-reference/supported-bindings/sendgrid.md (hunk 1 above)
Comment (yannart) [reply daprdocs/content/en/reference/components-reference/supported-bindings/sendgrid.md]: Ok, good suggestion

**Comment by yannart** [comment]
> @yannart - Is this targeting the v1.12 release ( in the code). If so then can you point this to the v1.12 docs branch?

Yes, the code is tagged to be in v1.12. However I do not see such doc branch.

**Comment by yannart** [comment]
@msfussell Thanks for the review, I attended your suggestions. Please just le me know how I can target the branch v1.12 (or feel free to modify the PR)

File: daprdocs/content/en/reference/components-reference/supported-bindings/sendgrid.md [hunk 3]
```diff
@@ -78,6 +83,20 @@ You can specify any of the optional metadata properties on the output binding re
...
+{
+  "operation": "create",
+  "metadata": {
+    "emailTo": "changeme@example.net",
+    "subject": "An template email from Dapr SendGrid binding",
+    "dynamicTemplateId": "d-123456789",
+    "dynamicTemplateData": "{\"customer\":{\"name\":\"John Smith\"}}"
+  }
+}
+
```
Comment (ItalyPaleAle) [inline daprdocs/content/en/reference/components-reference/supported-bindings/sendgrid.md:100]: Missing ``` to close

Comment (yannart) [reply daprdocs/content/en/reference/components-reference/supported-bindings/sendgrid.md:100]: Thanks @ItalyPaleAle. Commit with fix done.

**Comment by hhunter-ms** [comment]
> @msfussell Thanks for the review, I attended your suggestions. Please just le me know how I can target the branch v1.12 (or feel free to modify the PR)

@yannart I believe the 1.12 preview branch will be ready once 1.11 is released - correct me if I'm wrong @greenie-msft

**Comment by yannart** [comment]
Changed target to v1.12. @msfussell @ItalyPaleAle Please check if ok for you.

**Review by hhunter-ms — APPROVED** [review]

**Pull request merged**